libraries/AP_OSD/AP_OSD.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * AP_OSD partially based on betaflight and inav osd.c implemention.
+ * clarity.mcm font is taken from inav configurator.
+ * Many thanks to their authors.
+ */
+
+#include "AP_OSD.h"
+#include "AP_OSD_MAX7456.h"
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/Util.h>
+
+#include <utility>
+
+const AP_Param::GroupInfo AP_OSD::var_info[] = {
+
+    // @Param: _TYPE
+    // @DisplayName: OSD type
+    // @Description: OSD type
+    // @Values: 0:None,1:MAX7456
+    // @User: Standard
+    AP_GROUPINFO_FLAGS("_TYPE", 1, AP_OSD, osd_type, 0, AP_PARAM_FLAG_ENABLE),
+
+    // @Param: _UPDATE_FONT
+    // @DisplayName: Update font
+    // @Description: Undate font inside osd chip
+    // @Values: 0:Do not update,1:Update
+    // @User: Standard
+    AP_GROUPINFO("_UPDATE_FONT", 2, AP_OSD, update_font, 1),
+
+    // @Group: 1_
+    // @Path: AP_OSD_Screen.cpp
+    AP_SUBGROUPINFO(screen[0], "1_", 3, AP_OSD, AP_OSD_Screen),
```

Comment:
in other drivers, such as GPS, the first index has no number and the additional ones do. Example:
OSD_xxxx
OSD2_xxxx
OSD3_xxxx
OSD4_xxxx

Reply:
that usually because we start out supporting only a single one. In this case I think its better to use OSD1

Reply:
Personally, I'd like the first one as OSD_. Maybe we can ask the group at the dev call?